src/pudl_archiver/zenodo/entities.py

```diff
@@ -13,13 +14,13 @@
 class Doi(ConstrainedStr):
```

Comment:
For some reason, when creating `DatasetSettings` objects I was getting validation errors. Making this explicitly compiled regex patterns fixed the problem for me. Also added unit tests to test this.